hack/common.sh

```diff
@@ -159,9 +184,16 @@ os::build::build_binaries() {
     for platform in "${platforms[@]}"; do
       os::build::set_platform_envs "${platform}"
       echo "++ Building go targets for ${platform}:" "${targets[@]}"
-      go install "${goflags[@]:+${goflags[@]}}" \
-          -ldflags "${version_ldflags}" \
-          "${binaries[@]}"
+      for artifact in ${binaries[@]}; do
```

Comment:
if the whole implementation of building the e2e test artifact is different, this probably should be a separate function, maybe even a separate script (build-e2e.sh, modeled after a combination of build-go.sh and test-integration.sh?)